src/main/java/net/fabricmc/tinyremapper/AsmClassRemapper.java

```diff
@@ -516,11 +531,28 @@ private static String getIndexName(int index, boolean plural) {
 			}
 		}
 
-		private static boolean isValidJavaIdentifier(String s) {
-			if (s == null || s.isEmpty()) return false;
+		private enum InvalidIdentifierReason {
+			INVALID,
+			KEYWORD
+		}
+
+		/**
+		 * @param s
+		 *        Name of the identifier
+		 * @return
+		 *        Reason why the identifier is invalid, or {@code null} if it is valid
+		 */
+		private static InvalidIdentifierReason getInvalidJavaIdentifierReason(String s) {
```

Comment:
This method is likely better off being split into something like isValidOrKeyword and isKeyword without the enum. Most uses should be fine with checking invalid once, then only isKeyword later (e.g. lines 466, 475).

Reply:
Tried to apply that in the new version